### PR TITLE
🔧: create pyproject.toml

### PR DESCRIPTION
**Pull request by nstarman**
Good for squash merging.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  readme = "README.md"
+  requires-python = ">=3.10"
+  license.file = "LICENSE"
+  authors = [
+    {name="Consortium for Python Data API Standards", email="data-apis@users.noreply.github.com"},
+    {name="Joren Hammudoglu", email="jhammudoglu@gmail.com"},
+    {name="Nathaniel Starkman", email="nstarman@users.noreply.github.com"}
+  ]
+  classifiers = [
+    "Development Status :: 3 - Alpha",
```
Comment (jorenham) [inline pyproject.toml:14]: I suppose that once we publish the first version, we'll be calling it an "alpha".
But perhaps the most "agile" thing would be to ignore the future and use something that's closer to our current project status, such as

```suggestion
    "Development Status :: 1 - Planning",
```

So I guess it's mostly personal preference, and in this case I don't have one 🤷🏻

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python",
+    "Topic :: Scientific/Engineering",
+    "Typing :: Typed",
+  ]
+  dependencies = []
+
+  [project.urls]
+    Changelog     = "https://github.com/data-apis/array-api-typing/releases"
+    Homepage      = "https://github.com/data-apis/array-api-typing"
```
Comment (jorenham) [inline pyproject.toml:32]: how about

```suggestion
    Repository      = "https://github.com/data-apis/array-api-typing"
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,108 @@
+[project]
+  name="array-api-typing"
```
Comment (jorenham) [inline pyproject.toml:2]:
```suggestion
  name = "array-api-typing"
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      "ISC001",  # Conflicts with formatter
+    ]
+  
+    [tool.ruff.lint.isort]
+        combine-as-imports = true
+        extra-standard-library = ["typing_extensions"]
+        known-local-folder = ["array_api_typing"]
+
+  [tool.ruff.format]
+    docstring-code-format = true
```
Comment (jorenham) [inline pyproject.toml:108]: It's probably cleaner to move the installation and configuration of the tools to separate PR's (even though at first glance this config looks very reasonable to me)

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,108 @@
+[project]
+  name="array-api-typing"
+  dynamic = ["version"]
```
[the remark this answers is not in the document]
Comment (nstarman) [reply pyproject.toml:3]: Hatch VCS handles the version. 
https://hatch.pypa.io/latest/config/metadata/#dynamic

[the remark this answers is not in the document]
Comment (jorenham) [reply pyproject.toml:3]: that's pretty neat

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  description = "Static typing support for the array API standard"
+  readme = "README.md"
+  requires-python = ">=3.10"
+  license.file = "LICENSE"
```
[the remark this answers is not in the document]
Comment (nstarman) [reply pyproject.toml:7]: SGTM.

File: pyproject.toml (hunk 3 above)
Comment (nstarman) [reply pyproject.toml]: Done.

File: pyproject.toml (hunk 1 above)
Comment (nstarman) [reply pyproject.toml]: Done.

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,108 @@
+[project]
+  name="array-api-typing"
+  dynamic = ["version"]
+  description = "Static typing support for the array API standard"
+  readme = "README.md"
```
[the remark this answers is not in the document]
Comment (nstarman) [reply pyproject.toml:5]: Done.

File: pyproject.toml (hunk 2 above)
Comment (nstarman) [reply pyproject.toml:32]: SGTM.

Comment (nstarman) [reply pyproject.toml]: DOne.

File: pyproject.toml (hunk 4 above)
Comment (nstarman) [reply pyproject.toml]: Done.

**Review by jorenham — APPROVED** [review]

**Pull request merged**